### PR TITLE
[Fix] babel's "loose mode" class transform enbrittles BufferList

### PR DESCRIPTION
**Pull request by ljharb**
```js
Object.defineProperty(Object.prototype, util.inspect.custom, {
  set(v) { throw 'this should not happen inside readable-stream'; }
});
```

With this change, I believe the output will use [[Define]] instead of [[Set]] for https://github.com/nodejs/node/blob/c101251a95cc82142bee4637f8db6cc360a06d82/lib/internal/streams/buffer_list.js#L167, and thus no longer fail when Object.prototype is modified.

### CODE REVIEW

**Comment by ljharb** [comment]
Seems like https://github.com/vweevers/readable-stream/commit/432d4b77439a8a825d7a330c2d16410e9ac15fbf is when it first was put into loose mode, in #299, due to #293. 

Does readable-stream still support pre-ES5 environments?

**Comment by mcollina** [comment]
We support IE11 in readable-stream v3, while we still support pre-es5 environments in v2. This can land but not be backported.

**Comment by mcollina** [comment]
Can you also regenerate the code?

```
cd build
node build.js v10.10.0
```

**Comment by ljharb** [comment]
just v3 is fine; I’m mainly concerned with the version of this that’s vendored in core; that one doesn’t need to run Babel at all :-)

will update the build shortly

File: lib/internal/streams/buffer_list.js [hunk 1]
```diff
@@ -22,170 +28,183 @@ module.exports =
...
+        customInspect: false
+      }));
     }
-
-    this.length -= c;
-    return ret;
-  } // Make sure the linked list only shows the minimal necessary information.
-  ;
-
-  _proto[custom] = function (_, options) {
```
Comment (ljharb) [inline lib/internal/streams/buffer_list.js]: this was the vulnerability to Object.prototype modification

File: lib/internal/streams/buffer_list.js [hunk 2]
```diff
@@ -22,170 +28,183 @@ module.exports =
...
+        ++c;
       }
 
-      ++c;
+      this.length -= c;
+      return ret;
+    } // Make sure the linked list only shows the minimal necessary information.
+
+  }, {
+    key: custom,
```
Comment (ljharb) [inline lib/internal/streams/buffer_list.js]: this is the `custom` method

File: lib/internal/streams/buffer_list.js [hunk 3]
```diff
@@ -22,170 +28,183 @@ module.exports =
...
-  };
-
-  _proto.join = function join(s) {
-    if (this.length === 0) return '';
-    var p = this.head;
-    var ret = '' + p.data;
-
-    while (p = p.next) {
-      ret += s + p.data;
+  _createClass(BufferList, [{
```
Comment (ljharb) [inline lib/internal/streams/buffer_list.js]: createClass uses Define internally, not Set

File: .babelrc [hunk 4]
```diff
@@ -13,11 +13,5 @@
       ],
       "debug": true
     }]
-  ],
-  "plugins": [
-    // Included in the preset, but we can't specify per-plugin options there
-    ["@babel/plugin-transform-classes", {
```
Comment (mcollina) [inline .babelrc]: Can you drop this dependency from package.json?

File: lib/internal/streams/buffer_list.js (hunk 3 above)
Comment (mcollina) [reply lib/internal/streams/buffer_list.js]: Great!

File: .babelrc (hunk 4 above)
Comment (ljharb) [reply .babelrc]: oh sure. It’s already included via the env preset but I’ll remove it anyways

Comment (ljharb) [reply .babelrc]: actually, there never was an explicit dep here (oops), it just worked by accident because env brings it in.

Comment (mcollina) [reply .babelrc]: Ooops.

**Comment by mcollina** [comment]
cc @vweevers what do you think?

**Comment by vweevers** [comment]
👍

**Comment by ljharb** [comment]
yay! what are the next steps to get this into core?

**Comment by mcollina** [comment]
I think you'll have to wait for npm to do a release, and then update core to that. Or you can force a replacement directly.

**Comment by ljharb** [comment]
done in https://github.com/npm/cli/issues/824 and https://github.com/nodejs/node/pull/31977

**Pull request merged**